### DISCUSSION TITLE
Failed to start in production using Yarn PnP

### DISCUSSION THREAD

**Discussion by 0xHexE** [vikejs/vike#948] [Discussion]
I am trying to run vite-plugin-ssr in production it is not working so far.

```
NODE_ENV=production PORT=3000 node --experimental-specifier-resolution=node dist-server/index.js --logLevel warn
```

If I don't mention NODE_ENV it works.

```
Error: Qualified path resolution failed: we looked for the following paths, but none could be accessed.

Source path: /home/horizon/workspace/anthaathi-next/.yarn/unplugged/baseui-virtual-49214431c6/node_modules/baseui/typography
Not found: /home/horizon/workspace/anthaathi-next/.yarn/unplugged/baseui-virtual-49214431c6/node_modules/baseui/typography

    at makeError (/home/horizon/workspace/anthaathi-next/.pnp.cjs:25250:34)
    at resolveUnqualified (/home/horizon/workspace/anthaathi-next/.pnp.cjs:27072:13)
    at resolveRequest (/home/horizon/workspace/anthaathi-next/.pnp.cjs:27113:14)
    at Object.resolveRequest (/home/horizon/workspace/anthaathi-next/.pnp.cjs:27169:26)
    at resolve$1 (file:///home/horizon/workspace/anthaathi-next/.pnp.loader.mjs:1989:21)
    at nextResolve (node:internal/modules/esm/hooks:726:28)
    at Hooks.resolve (node:internal/modules/esm/hooks:235:30)
    at handleMessage (node:internal/modules/esm/worker:168:24)
    at Immediate.checkForMessages (node:internal/modules/esm/worker:117:28)
    at process.processImmediate (node:internal/timers:478:21)

```

**Comment by 0xHexE** [comment]
Weird is `/home/horizon/workspace/anthaathi-next/.yarn/unplugged/baseui-virtual-49214431c6/node_modules/baseui/typography` this path exists. I belive it is because of the extension is required.

![image](https://github.com/brillout/vite-plugin-ssr/assets/29942725/12b8b741-1a03-4e1f-995b-bd0646342b3c)

**Comment by 0xHexE** [comment]
Finally, I write my own loader and it worked. I am not sure why this happened.

```typescript
export function resolve(specifier, context, nextResolve) {
	if (specifier.startsWith("baseui/")) {
		return nextResolve(specifier + '/index.js');
	}

	return nextResolve(specifier);
}
```

**Reply by brillout** [reply]
So it turned out to be unrelated to VPS, right?

**Reply by 0xHexE** [reply]
I believe it is related to VPS. This is more like a hack. I am going through the source code of the VPS If I get it somewhere I'll get back.

**Reply by 0xHexE** [reply]
The package is being treated as a folder so this is happening I think. We need to use PnP API to resolve this properly.

**Reply by 0xHexE** [reply]
https://vite-plugin-ssr.com/broken-npm-package#solution

I think this is the issue. I have to mention noExternal I think.

**Reply by 0xHexE** [reply]
But I am thinking we can improve this because it works in dev just happens when we run on a production run with `NODE_ENV=production` So there is scope to improve.

**Reply by brillout** [reply]
@0xHexE I think the best would be a reproduction, so I can get my hands on it. (FYI it's more of an ecosystem issue, not really a VPS one. But VPS may workaround it, let's see.)

**Reply by 0xHexE** [reply]
This is the reproduction https://github.com/0xHexE/vps-d-948.git.

Try this steps
```bash
yarn install
yarn build
yarn server:prod # and visit site
yarn server:prod-dev # and visit site
```

And use node version 19 I believe older version will not support this

**Reply by 0xHexE** [reply]
Hi @brillout, Just a friendly nudge.

**Reply by brillout** [reply]
I've actually had a look at it: it really seems like a Yarn problem. I don't think it's in any way related to VPS nor Vite?

**Reply by 0xHexE** [reply]
I am aware that working with Yarn PnP can be challenging at times. Please allow me to examine the issue and determine if I can rectify it. I will then provide you with an update.

**Reply by brillout** [reply]
👍 Also definitely let me know if you believe there is something vite-plugin-ssr can do to improve the situation.

**Reply by RobIsHere** [reply]
The flag --experimental-specifier-resolution has been removed in node 18 and is probably ignored by node 19. The option to specify an own loader is according to their docs the new way to go - just what you did. (no Expert, just found out a lot trying to resolve my own problems with ssr)
